webauthn.go

```diff
@@ -0,0 +1,339 @@
+		// unknown values, treating an unknown value as if the member does not exist. Eligible authenticators are
+		// filtered to only those capable of satisfying this requirement.
+		UserVerification string `json:"userVerification"`
+		// This OPTIONAL member contains additional parameters requesting additional processing by the client and
+		// authenticator. For example, if transaction confirmation is sought from the user, then the prompt string
+		// might be included as an extension.
+		Extensions map[string]interface{} `json:"extensions"`
+	}
+
+	// The PublicKeyCredentialRpEntity is used to supply additional Relying Party attributes when creating a new
+	// credential.
+	PublicKeyCredentialRpEntity struct {
+		// A unique identifier for the Relying Party entity, which sets the RP ID.
+		ID string `json:"id"`
+		// A human-palatable identifier for the Relying Party, intended only for display.
+		Name string `json:"name"`
+	}
+
+	// The PublicKeyCredentialUserEntity is used to supply additional user account attributes when creating a new
+	// credential.
+	PublicKeyCredentialUserEntity struct {
+		// The user handle of the user account entity. A user handle is an opaque byte sequence with a maximum size of
+		// 64 bytes, and is not meant to be displayed to the user.
+		//
+		// To ensure secure operation, authentication and authorization decisions MUST be made on the basis of this id
+		// member, not the displayName nor name members.
+		//
+		// The user handle MUST NOT contain personally identifying information about the user, such as a username or
+		// e-mail address.
+		ID []byte `json:"id"`
+		// A human-palatable name for the user account, intended only for display. For example, "Alex Müller" or
+		// "田中倫". The Relying Party SHOULD let the user choose this, and SHOULD NOT restrict the choice more than
+		// necessary.
+		//
+		// Authenticators MUST accept and store a 64-byte minimum length for a displayName member’s value.
+		// Authenticators MAY truncate a displayName member’s value so that it fits within 64 bytes.
+		DisplayName string `json:"displayName"`
+		// A human-palatable identifier for a user account.
+		Name string `json:"name"`
+	}
+
+	// A TokenBinding is established by a User Agent generating a private-public key pair (possibly within a secure
+	// hardware module, such as a Trusted Platform Module) per target server, providing the public key to the server,
+	// and proving possession of the corresponding private key.
+	TokenBinding struct {
+		// This member SHOULD be a member of TokenBindingStatus but client platforms MUST ignore unknown values,
+		// treating an unknown value as if the tokenBinding member does not exist.
+		Status string `json:"status"`
+		// This member MUST be present if status is present, and MUST be a base64url encoding of the Token Binding ID
+		// that was used when communicating with the Relying Party.
+		ID string `json:"id"`
+	}
+)
+
+const (
```

Comment:
nit: could we split these into separate `const()` blocks by type?

eg...

```
// Link to the docs for all the enumy-types
const(
// AttestationA does ....
AttestationA = ...
AttestationB = ...
AttestationC = ...
)

const(
COSEAlgorithmIdentifierX=foo
COSEAlgorithmIdentifierY=foo
COSEAlgorithmIdentifierZ=foo
)
```
etc